cpp/example_code/s3-crt/s3-crt-demo.cpp

```diff
@@ -0,0 +1,222 @@
+#include <aws/core/utils/logging/CRTLogSystem.h>
+#include <aws/s3-crt/S3CrtClient.h>
+#include <aws/s3-crt/model/CreateBucketRequest.h>
+#include <aws/s3-crt/model/DeleteBucketRequest.h>
+#include <aws/s3-crt/model/PutObjectRequest.h>
+#include <aws/s3-crt/model/GetObjectRequest.h>
+#include <aws/s3-crt/model/DeleteObjectRequest.h>
+
+static const char ALLOCATION_TAG[] = "s3-crt-demo";
+
+// List all Amazon S3 buckets under the account.
+bool ListBuckets(const Aws::S3Crt::S3CrtClient& s3CrtClient, const Aws::String& bucketName) {
+
+    Aws::S3Crt::Model::ListBucketsOutcome outcome = s3CrtClient.ListBuckets();
+
+    if (outcome.IsSuccess()) {
+        std::cout << "All buckets under my account:" << std::endl;
+
+        for (auto const& bucket : outcome.GetResult().GetBuckets())
+        {
+            std::cout << "  * " << bucket.GetName() << std::endl;
+        }
+        std::cout << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "ListBuckets error:\n"<< outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// Create an Amazon S3 bucket.
+bool CreateBucket(const Aws::S3Crt::S3CrtClient& s3CrtClient, const Aws::String& bucketName) {
+
+    std::cout << "Creating bucket: \"" << bucketName << "\" ..." << std::endl;
+
+    Aws::S3Crt::Model::CreateBucketRequest request;
+    request.SetBucket(bucketName);
+
+    Aws::S3Crt::Model::CreateBucketOutcome outcome = s3CrtClient.CreateBucket(request);
+
+    if (outcome.IsSuccess()) {
+        std::cout << "Bucket created." << std::endl << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "CreateBucket error:\n" << outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// Delete an existing Amazon S3 bucket.
+bool DeleteBucket(const Aws::S3Crt::S3CrtClient& s3CrtClient, const Aws::String& bucketName) {
+
+    std::cout << "Deleting bucket: \"" << bucketName << "\" ..." << std::endl;
+
+    Aws::S3Crt::Model::DeleteBucketRequest request;
+    request.SetBucket(bucketName);
+
+    Aws::S3Crt::Model::DeleteBucketOutcome outcome = s3CrtClient.DeleteBucket(request);
+
+    if (outcome.IsSuccess()) {
+        std::cout << "Bucket deleted." << std::endl << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "DeleteBucket error:\n" << outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// Put an Amazon S3 object to the bucket.
+bool PutObject(const Aws::S3Crt::S3CrtClient& s3CrtClient, const Aws::String& bucketName, const Aws::String& objectKey) {
+
+    std::cout << "Putting object: \"" << objectKey << "\" to bucket: \"" << bucketName << "\" ..." << std::endl;
+
+    Aws::S3Crt::Model::PutObjectRequest request;
+    request.SetBucket(bucketName);
+    request.SetKey(objectKey);
+    auto bodyStream = Aws::MakeShared<Aws::StringStream>(ALLOCATION_TAG);
+    *bodyStream << "s3-crt-demo";
+    request.SetBody(bodyStream);
```

Comment:
[opinion] The [existing s3 demo](https://github.com/awsdocs/aws-doc-sdk-examples/blob/8ed203647cd1f39c3df4630d5e9eb1994a00ab11/cpp/example_code/s3/put_object.cpp#L38) uploads a file from disk. That seems more interesting than uploading the string "s3-crt-demo"